### ISSUE TITLE
Conflict between Polymer (webcomponents.js) and Highchart (highcharts.js) 

### ISSUE THREAD

**Issue by hcho112** [highcharts/highcharts#3775]
To work with Google Polymer and Highcharts in IE environment, it breaks. 
Detailed descriptions are posted on stackoverflow:

http://stackoverflow.com/questions/28205698/conflict-between-google-polymer-and-highcharts-library

Live demo is loaded on jsfiddle:

http://jsfiddle.net/epavak4y/2/

Only having an issue when it is tested in IE. It seems like webcomponents.js does overwrite Document.querySelector function that may end up impact on Highcharts library. However issue is only occurred in IE environment. If it is something that Highcharts are not responsible of, at least I would like to know where about within source code that are responsible for rendering labels and hover so I can investigate the issue more.

Thank you for your support.

**Comment by highslide-software** [comment]
For some reason IE is not able to calculate the bounding box of text elements in this case. The problem arises on this line: https://github.com/highslide-software/highcharts.com/blob/master/js/parts/SvgRenderer.js#L616 .

An error is thrown on `element.getBBox()`, and the message is "Invalid calling object". You can debug this by adding a `console.log(e.message)` inside the catch statement. 

Why this happens is another case. The element is an SVG text element, and other browsers have no problem with it. You could take this back to the Polymer developers so they can take it further.

For the records, I have tried rendering Highcharts to hidden or disconnected nodes, but that too works well, so it is not the reason why the elements don't have a bounding box: http://jsfiddle.net/highcharts/721to3sj/

**Comment by hcho112** [comment]
After waiting for response from Polymer developers, I just went to debug by myself and made hotfix. I'm not sure if it is a practical solution.. but it works. 
Before to present hotfix code, I would like to describe some of things that I found during debug. 

1. Current issue only happens in IE. 
2. After webcomponents.js polyfill gets applied, it masses with Jquery's .offset() and .position() functions. It basically returns same output. (or at least on my case)
3. Render got massed up because part of 'element.getBBox()' function is to get 'bBox' variable with defined width and height. (could be more attr..) But it returns with '{ width: 0, height: 0 };' in IE only. I assumed that these attributes are responsible for locating element. 

Then I saw a someone had a similar issue and resolved by extending Jquery offset function by re-defining 'jQuery.fn.offset' function with using '.getBoundingClientRect()' which is native browser Javascript function to get width, height, top, bottom.. etc etc. So I thought, I might be able to hack around with .getBoundingClientRect function. so I tested in Chrome to see if value of 'element.offsetWidth' and 'element.offsetHeight' (which becomes bBox.width and bBox.height) is similar or same if I use 'element.getBoundingClientRect()' to get width and height. It turns out they are. So Here the the hot fix code that I wrote:

---------------------------------------------------------------------------------------------------
        if(element.offsetWidth == undefined && element.offsetHeight == undefined ) {
                bBox.width = element.getBoundingClientRect().width;
                bBox.height = element.getBoundingClientRect().height;
        }

        return bBox;
---------------------------------------------------------------------------------------------------

Just before bBox get returned, check if width and height is undefined. (This only happens on IE with polyfill) then reset with ' element.getBoundingClientRect()' function. 

Turns out that it does solve my issue. Personally I wouldn't want to alter source code but I couldn't found any other way.
